enabler/src/de/schildbach/pte/VrsProvider.java

```diff
@@ -0,0 +1,909 @@
+					long distance = segment.has("distance") ? segment.getLong("distance") : 0;
+					Line line = null;
+					String direction = null;
+					if (segment.has("line")) {
+						JSONObject lineObject = segment.getJSONObject("line");
+						line = parseLine(lineObject);
+						if (lineObject.has("direction")) {
+							direction = lineObject.getString("direction");
+						}
+					}
+					StringBuilder message = new StringBuilder();
+					if (segment.has("infos")) {
+						JSONArray infos = segment.getJSONArray("infos");
+						for (int k = 0; k < infos.length(); k++) {
+							if (k > 0) {
+								message.append(", ");
+							}
+							message.append(infos.getJSONObject(k).getString("text"));
+						}
+						// System.out.println("Message is " + message.toString());
+					}
+
+					List<Point> points = new ArrayList<Point>();
+					points.add(new Point(segmentOrigin.lat, segmentOrigin.lon));
+					for (Stop intermediateStop : intermediateStops) {
+						points.add(new Point(intermediateStop.location.lat, intermediateStop.location.lon));
+					}
+					points.add(new Point(segmentDestination.lat, segmentDestination.lon));
+					if (type.equals("walk")) {
+						if (departurePlanned == null) {
+							departurePlanned = legs.get(j - 1).getArrivalTime();
+						}
+						if (arrivalPlanned == null) {
+							arrivalPlanned = new Date(legs.get(j - 1).getArrivalTime().getTime() + traveltime * 1000);
+						}
+						legs.add(new Trip.Individual(Trip.Individual.Type.WALK, segmentOrigin, departurePlanned, segmentDestination, arrivalPlanned, points, (int) distance));
+						// System.out.println("walk from " + segmentOrigin + "//" + segmentOriginPosition + " at "+ departurePlanned + " to " + segmentDestination + "//" + segmentDestinationPosition + " at " + arrivalPlanned);
+					} else if (type.equals("publicTransport")) {
+						legs.add(new Trip.Public(line, direction != null ? new Location(LocationType.STATION, null /* id */, null /* place */, direction) : null,
+								new Stop(segmentOrigin, true /* departure */, departurePlanned, departurePredicted, segmentOriginPosition, segmentOriginPosition),
+								new Stop(segmentDestination, false /* departure */, arrivalPlanned, arrivalPredicted, segmentDestinationPosition, segmentDestinationPosition),
+								intermediateStops, points, message.toString()));
+						// System.out.println("ride from " + segmentOrigin + "//" + segmentOriginPosition + " at "+ departurePlanned + " to " + segmentDestination + "//" + segmentDestinationPosition + " at " + arrivalPlanned + " taking " + (line == null ? "null" : line) + " direction " + (direction ==  null ? "null" : direction));
+					}
+				}
+				int changes = route.getInt("changes");
+				List<Fare> fares = new ArrayList<Fare>();
+				if (route.has("costs")) {
+					final JSONObject costs = route.getJSONObject("costs");
+
+					final String name = costs.has("name") ? costs.getString("name") : null; // seems constant "VRS-Tarif"
```

Comment:
You can use costs.optString() here, to get rid of the conditional. [multiple occurences]